### ISSUE TITLE
Workspace: Previously selected dataset collection not remembered

### ISSUE THREAD

**Issue by jorvis** [IGS/gEAR#744]
This is set in the user's profile so it auto-selects the next time, but this currently isn't happening.

**Comment by adkinsrs** [comment]
I may have fixed this in my updates for #715

**Comment by jorvis** [comment]
In my local testing this is still an issue

**Comment by adkinsrs** [comment]
Ah.  I had added an extra statement to `fetchDatasetCollections` to select the "selected_dc_share_id" when `get_user_layouts.cgi` is run.

**Comment by jorvis** [comment]
You 'had', as in I accidentally removed it or something?  If so - oops.

**Comment by adkinsrs** [comment]
There are also code snippets like this in index.js, expression.js, and projection.js

https://github.com/IGS/gEAR/blob/2fc0022ef64030ab371a89c3c4e3c959c1585cb8/www/js/index.js#L169-L171

As long as the user is logged in, the gear_default_domain cookie should populate the CURRENT_USER.default_profile_share_id

**Comment by jorvis** [comment]
So I've ended up really overhauling how this works - something I've wanted to do for a while.  This will ultimately get rid of the layout.is_current column and the 'gear_default_domain' cookie we've used and at the same time allow us to specify a site-specific default layout in the gear.ini file.  Much of this is committed already at this point, but now there's a new guser.layout_share_id column to store the most recently selected one.  This has been propagated to both the python API and JS object layers.

The only thing left to do still is actually save it in the DB when the user changes any layout (dataset collection).  Another good reason to make each selector on the page use the widget where appropriate.
